addons/web/static/src/core/py_js/py_date.js

```diff
@@ -580,24 +639,70 @@ export class PyRelativeDelta {
      * @returns {PyDateTime}
      */
     static substract(date, delta) {
+        date = delta.applyAbsolute(date);
+        const leapDays = delta.getLeapDays(date);
+
         const s = tmxxx(
             date.year,
             date.month,
-            date.day - delta.day,
+            date.day - delta.day + leapDays,
             -delta.hour,
             -delta.minute,
             -delta.second
         );
+
+        if (delta.weekday !== null) {
+            const sprime = delta.applyWeekday(s);
+            Object.assign(s, sprime);
+        }
+
         return new PyDateTime(s.year, s.month, s.day, s.hour, s.minute, s.second, 0);
     }
 
+    applyAbsolute(oldDdate) {
+        const date = new PyDateTime();
+        Object.assign(date, oldDdate);
+        for (const absoluteName of this.absolute) {
+            date[absoluteName] = this[absoluteName];
+            this[absoluteName] = 0;
+        }
+        return date;
```

Comment:
also isinstance(datetime.date(2021,9,30) + rd(days=+1),datetime.date) is True in python.
Maybe use a timedelta and (date/datetime).add(timedelta) as before?

Reply:
isinstance(datetime.date(2021,9,30) + relativedelta(days=+1), datetime.datetime) is False ^^